components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/kafka/KAFKAInjectHandler.java

```diff
@@ -0,0 +1,145 @@
+import org.apache.axis2.builder.Builder;
+import org.apache.axis2.builder.BuilderUtil;
+import org.apache.axis2.builder.SOAPBuilder;
+import org.apache.axis2.context.MessageContext;
+import org.apache.axis2.transport.TransportUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.mediators.base.SequenceMediator;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * KafkaInjectHandler uses to mediate the received Kafka message
+ */
+public class KAFKAInjectHandler implements InjectHandler {
+    private static final Log log = LogFactory.getLog(KAFKAInjectHandler.class);
+
+    private String injectingSeq;
+    private String onErrorSeq;
+    private boolean sequential;
+    private SynapseEnvironment synapseEnvironment;
+    private String contentType;
+
+    public KAFKAInjectHandler(String injectingSeq, String onErrorSeq,boolean sequential,
+                              SynapseEnvironment synapseEnvironment, String contentType) {
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.sequential=sequential;
+        this.synapseEnvironment = synapseEnvironment;
+        this.contentType = contentType;
+    }
+
+    /**
+     * Determine the message builder to use, set the message payload to the message context and
+     * inject the message to the sequence
+     */
+    public boolean invoke(Object object) {
+        byte[] msg = (byte[]) object;
+
+        org.apache.synapse.MessageContext msgCtx = createMessageContext();
+        log.debug("Processed Kafka Message ");
+        MessageContext axis2MsgCtx = ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx)
+                .getAxis2MessageContext();
+        // Determine the message builder to use
+        Builder builder = null;
+        if (contentType == null) {
+            log.debug("No content type specified. Using SOAP builder.");
+            builder = new SOAPBuilder();
+        } else {
+            int index = contentType.indexOf(';');
+            String type = index > 0 ? contentType.substring(0, index)
+                    : contentType;
+            try {
+                builder = BuilderUtil.getBuilderFromSelector(type, axis2MsgCtx);
+            } catch (AxisFault axisFault) {
+                log.error("Error while creating message builder :: "
+                        + axisFault.getMessage());
+
+            }
+            if (builder == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("No message builder found for type '" + type
+                            + "'. Falling back to SOAP.");
+                }
+                builder = new SOAPBuilder();
+            }
+        }
+        OMElement documentElement = null;
+        // set the message payload to the message context
+        InputStream in = new ByteArrayInputStream(msg);
```

Comment:
InputStream in = new AutoCloseInputStream(new ByteArrayInputStream(msg));
Fixed in the PR - https://github.com/wso2/carbon-mediation/pull/451